python/seldon_core/batch_processor.py

```diff
@@ -316,23 +320,102 @@ def _send_batch_predict_multi_request(
     """
 
     indexes = [x[0] for x in input_data]
-    instance_ids = [x[1] for x in input_data]
+
+    seldon_puid = input_data[0][1]
+    instance_ids = [f"{seldon_puid}-item-{n}" for n, _ in enumerate(input_data)]
+    loaded_data = [json.loads(data[2]) for data in input_data]
 
     predict_kwargs = {}
     tags = {
         "batch_id": batch_id,
     }
-
     predict_kwargs["meta"] = tags
-    predict_kwargs["headers"] = {"Seldon-Puid": instance_ids[0]}
+    predict_kwargs["headers"] = {"Seldon-Puid": seldon_puid}
 
     try:
```

Comment:
Feels like all this try block could be extracted into separate function as it's quite extensive

Reply:
done